ksqldb-engine/src/main/java/io/confluent/ksql/function/udaf/max/TimeMaxKudaf.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2022 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.function.udaf.max;
+
+import io.confluent.ksql.function.udaf.BaseComparableKudaf;
+import io.confluent.ksql.schema.ksql.types.SqlTypes;
+import java.sql.Time;
+
+public class TimeMaxKudaf extends BaseComparableKudaf<Time> {
+
+  TimeMaxKudaf(
+      final String functionName,
+      final Integer argIndexInValue
+  ) {
+    super(functionName,
+        argIndexInValue,
+        SqlTypes.TIME,
+        (first, second) -> first.compareTo(second) > 0 ? first : second,
```

Comment:
This function ought to work for all the implementations.  I think we can push it into the `BaseComparableKudaf`.

Further, I'd suggest making concrete `Min` and `Max` versions of `BaseComparableKudaf` and deleting the 14 or so {Time, Long, Double, ...}{Max|Min}Kudaf classes.

Reply:
Yup, thats a good idea! Was playing safe keeping the existing classes intact